experimental/torch_xla2/test/test_unbounded_dynamism.py

```diff
@@ -0,0 +1,662 @@
+import re
+import sys
+import unittest
+
+import numpy as np
+import torch
+from torch.export import Dim, export
+from torch_xla2.export import exported_program_to_stablehlo as exp2shlo
+
+## This file is copied from `xla/test/stablehlo/test_unbounded_dynamism.py`
+## To test that torch_xla2 has identical behavior.
+## The only differences in this test files are that torch_xla2 export preserves
+## argument order more often than torch_xla export.
+##
+## This broke ~5 tests, for example: test_bmm_dynamic_out_dim
+##   args = (
+##     torch.rand((8, 128, 256)),
+##     torch.rand((8, 256, 3)),
+##   )
+##   dynamic_shapes = ((None, {2: Dim("dim")}),)
+##   ...
+##   torch_xla_regex = r'%arg.: tensor<8x256x\?xf32>.*%arg.: tensor<8x128x256xf32>.*->.*tensor<8x128x\?xf32>'
+##   torch_xla2_regex = r'%arg.: tensor<8x128x256xf32>.*%arg.: tensor<8x256x\?xf32>.*->.*tensor<8x128x\?xf32>'
+
+# Shim to run tests
+class ExportAdapter():
+  def __init__(self, export):
+    self.export = export
+
+  def get_stablehlo_text(self):
+    return self.export.mlir_module()
+
+def exported_program_to_stablehlo(exported):
+  return ExportAdapter(exp2shlo(exported))
+
+def wrap_func_as_nn_module(f):
+  class M(torch.nn.Module):
+    def __init__(self):
+      super().__init__()
+      
+    def forward(self, *args):
+      return f(*args)
+  return M().eval()
+
+class UnboundedDynamismExportTest(unittest.TestCase):
+
+  def test_add(self):
+    args = (torch.rand((10, 197, 768)), torch.rand((10, 197, 768)))
+    dynamic_shapes = (({0: Dim("dim")}, {0: Dim("dim")}),)
+    m = wrap_func_as_nn_module(torch.ops.aten.add.Tensor)
+    ep = export(m, args=args, dynamic_shapes=dynamic_shapes)
+    shlo_module = exported_program_to_stablehlo(ep)
+    shlo_text = shlo_module.get_stablehlo_text()
+    self.assertTrue(
+        re.search(
+            r'tensor<\?x197x768xf32>.*tensor<\?x197x768xf32>.*->.*tensor<\?x197x768xf32>',
+            shlo_text) is not None)
```

Comment:
Would `assertRegex` work here?

Reply:
This file is forked from a `test_unbounded_dynamism.py` file that exists in the main torch_xla tests (keeping separate since there's little overlap in testing currently). Want it to be as comparable as possible, so using this formatting. But yes, I use assertRegex elsewhere in the new tests